### PR TITLE
Feature/om 184

### PR DESCRIPTION
**Pull request by Citronhariom**
Descriptions : Build has been completed for the related story OM-184 and OM-410

### CODE REVIEW

File: force-app/main/default/classes/OpportunityValidationWrapperTest.cls [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+@isTest
+public with sharing class OpportunityValidationWrapperTest {
+
+    static testMethod void validateOpportunitiesBeforeInsertMethod() {
+        
+        test.startTest();
+        Group testGroup = new Group(Name = 'Customer_Success', type = 'Queue');
+        insert testGroup;
+        QueueSObject mappingObject = new QueueSobject(QueueId = testGroup.Id, SObjectType = 'Case');
+        System.runAs(new User(Id = UserInfo.getUserId())) {
```
Comment (MHigel) [inline force-app/main/default/classes/OpportunityValidationWrapperTest.cls:12]: Why are we doing this?  This code will always run as the running user, so using `System.runAs` doesn't really change the behavior, right?

File: force-app/main/default/classes/OpportunityValidationWrapperTest.cls [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        opp1.Name = acc.Name + ' Opportunity1 ';
+        opp1.AccountId = acc.Id;
+        opp1.StageName = 'Presentation';
+        opp1.CloseDate = System.today().addDays(30);
+        opp1.Amount = 0.0;
+        opp1.Type = 'Retention';
+        opp1.Opportunity_Type__c = 'Consumables';
+        opp1.LeadSource = 'Call In';
+        opp1.RecordTypeId = newRecordTypeId;
+        insert opp1;
```
Comment (MHigel) [inline force-app/main/default/classes/OpportunityValidationWrapperTest.cls:70]: We should combine the insert DML of these opps into a single DML using a collection.

File: force-app/main/default/classes/OpportunityValidationWrapperTest.cls [hunk 3]
```diff
@@ -0,0 +1,92 @@
+
+
+@isTest
+public with sharing class OpportunityValidationWrapperTest {
+
+    static testMethod void validateOpportunitiesBeforeInsertMethod() {
```
Comment (MHigel) [inline force-app/main/default/classes/OpportunityValidationWrapperTest.cls:6]: There are no asserts in this test, so we have no way of validating whether the tested functionality actually works.  Those asserts must validate the expected outputs and behaviors.

**Review by MHigel — CHANGES_REQUESTED** [review]

**Comment by Citronhariom** [comment]
This PR got multiple wrong commit , So creating new PR for the user story om-393

**Pull request closed without merging**